src/services/catalog.service.ts

```diff
@@ -0,0 +1,152 @@
+import * as _ from 'lodash';
+
+export class CatalogService {
+  static $inject = ['Constants', '$filter'];
+
+  public $filter: any;
+  private constants: any;
+  private categories: any;
+
+  constructor(constants: any, $filter: any, $scope: any) {
+    this.constants = constants;
+    this.$filter = $filter;
+    this.categories = this.constants.SERVICE_CATALOG_CATEGORIES;
+  }
+
+  public getServiceItem(resource: any) {
+    return new ServiceItem(resource, this);
+  }
+
+  public getImageItem(resource: any) {
+    return new ImageItem(resource, this);
+  }
+
+  public getCategoriesBySubCategories(tags: any) {
+    let catsBySubCats = {};
+    _.each(tags, (tag) => {
+      _.each(this.categories, (category) => {
+        let subCat: any = _.find(category.subCategories, (subCategory: any) => {
```

Comment:
you know what, just realized if this matches multiple subcategories only the first one will actually get matched, i dont think we can break out early at all here